crates/core/src/db/datastore/locking_tx_datastore/datastore.rs

```diff
@@ -154,9 +194,14 @@ impl Locking {
                 Operation::Insert => {
                     let (table, blob_store) =
                         committed_state.get_table_and_blob_store_or_create_ref_schema(table_id, &schema);
-                    table.insert(blob_store, &row).unwrap_or_else(|e| {
-                        panic!("Failed to insert during transaction playback: {:?}", e);
-                    });
+                    table
+                        .insert_internal_allow_duplicate(blob_store, &row)
```

Comment:
Is this indeed enough? This method doesn't update the pointer map, which seems important.